lib/dialects/abstract/query-generator.js

```diff
@@ -102,6 +102,22 @@ const QueryGenerator = {
       emptyQuery += ' VALUES ()';
     }
 
+    if (this._dialect.supports['ON CONFLICT']) {
+      if (options.onConflict && semver.gte(this.sequelize.options.databaseVersion, '9.5.0')) {
```

Comment:
You also have to check if dialect is `postgres`, as only `postgres@9.5` support `ON CONFLICT` in this case